### PR TITLE
RATEST-198: Adjust windows os configuration to make drivers for browsers executable

### PR DESCRIPTION
**Pull request by kdaud**
Ticket [ID](https://issues.openmrs.org/browse/RATEST-198)

Description -> Adjust windows os configuration to make drivers for browsers executable

### CODE REVIEW

**Comment by sherrif10** [comment]
Thanks @kdaudi, I wish an individual with Windows is can carry out this
configuration since it's going to be little tricky for you to clarify
whether things are passing or not. Let me know your suggestions

**Comment by kdaud** [comment]
I do carry out simulation on a windows os embedded in a virtual box. Well virtual windows os mimics real windows os!

**Comment by sherrif10** [comment]
Lest have one carrying out simulations on real windows thats a virtual window box , otherwise we wont be sure whether a fix has been resolved cc @herman

File: pom.xml [hunk 1]
```diff
@@ -87,7 +87,7 @@
                         <phase>process-test-classes</phase>
                         <configuration>
                             <target>
-                                <chmod file="target/test-classes/chromedriver/windows/chromedriver.exe" perm="755" />
+                                <chmod file="target/test-classes/chromedriver/windows/chromedriver" perm="755" />
```
Comment (sherrif10) [inline pom.xml:90]: This should be having an extenson .exe

**Comment by kdaud** [comment]
The changes have made have been [simulated](https://github.com/openmrs/openmrs-contrib-qaframework/pull/80#issuecomment-901613767) and that is why I created the ticket for it. The ticket title and the commit message tell it all.

**Comment by mherman22** [comment]
> Lets have one carrying out simulations on real windows thats a virtual window box , otherwise we wont be sure whether a fix has been resolved cc @herman

Let me test this an get back here!!!!

**Comment by kdaud** [comment]
@mherman22 ensure to download a chrome or firefox driver  for windows os matching your browser version installed on your pc.

**Comment by sherrif10** [comment]
You can simply checkout this branch from master and build locally

**Comment by mherman22** [comment]
i have tried checking it, i dont know if my environment is faulty or something, as discussed [here](https://talk.openmrs.org/t/build-failure-on-running-mvn-clean-install-on-openmrs-contrib-qaframework/34328?u=mherman22) , i still face the samething.
i need to check out my environment

**Comment by kdaud** [comment]
@mherman22 can we sync together for the test simulation ?

**Comment by sherrif10** [comment]
Hello @mherman22 , Any progress on testing this on windows machine

**Comment by kdaud** [comment]
@mherman22 seems to have some issues when it came to simulation. However, I personally took the time to get a real windows os environment and tested the changes and local build plan was passing the workflow

**Comment by mherman22** [comment]
> Hello @mherman22 , Any progress on testing this on windows machine

sorry @sherrif10 i opted for a linux os but im certain @kdaud tested the configurations on a real platform and they shoul work fine!

**Comment by kdaud** [comment]
cc: @dkayiwa

File: pom.xml (hunk 1 above)
Comment (dkayiwa) [reply pom.xml:90]: @kdaud did you respond to this comment?

Comment (kdaud) [reply pom.xml:90]: > This should be having an extension .exe

Once the driver is executable which certainly should be made an executable one, then we do not need to have an extension of `.exe` in our configurations

Here are the configurations fo linux and mac os

Mac os [chrome ](https://github.com/openmrs/openmrs-contrib-qaframework/blob/b60679a483fd6e18535b116070f2f2018434f35f/pom.xml#L64)and [firefox
](https://github.com/openmrs/openmrs-contrib-qaframework/blob/b60679a483fd6e18535b116070f2f2018434f35f/pom.xml#L103)

Linux os [chrome](https://github.com/openmrs/openmrs-contrib-qaframework/blob/b60679a483fd6e18535b116070f2f2018434f35f/pom.xml#L77) and [firefox](https://github.com/openmrs/openmrs-contrib-qaframework/blob/b60679a483fd6e18535b116070f2f2018434f35f/pom.xml#L116)

**Comment by sherrif10** [comment]
Seems all the discussion we had were not finished. Any folk testing this change will help us such that we are on the right track. However like I commented on slack. Let's try to use Linux specifically since that's the way future

**Comment by sherrif10** [comment]
@kdaud  there are some conflicting files

**Comment by kdaud** [comment]
@gracebish  could you tests out these changes on your windows machine and give a feedback here?.

**Comment by gracebish** [comment]
@kdaud after adding the changes in the pom.xml file of RATEST-302, i get success in all builds. https://snipboard.io/Sskg6G.jpg

**Comment by kdaud** [comment]
> @kdaud after adding the changes in the pom.xml file of RATEST-302, i get success in all builds. https://snipboard.io/Sskg6G.jpg

Could you provide the actual logs to the build?

**Comment by gracebish** [comment]
@kdaud here is the pastebin https://pastebin.com/ksBjeWBQ.

**Comment by kdaud** [comment]
@gracebish so now are you able to run any of the existing tests? For example

```js
 npm run refapp2PatientDemographics
```

**Comment by sherrif10** [comment]
can you also try to first build this module locally and push again , this have sometimes saved me from refapp 3.x builds.

**Comment by kdaud** [comment]
> Can you also try to first build this module locally and push again , this have sometimes saved me from refapp 3.x builds.

@sherrif10 are you directing the question to Grace?

**Comment by sherrif10** [comment]
> directing the question to Grace?

Sure . True

**Comment by sherrif10** [comment]
@kdaud  how with this executable on Windows machine. Is there any work around concerning this issue. Or we can add some resources encouraging folks to use Linux operating system.

**Comment by kdaud** [comment]
> @kdaud how with this executable on Windows machine. Is there any work around concerning this issue

The changes need to be tested on a windows environment! 

But will have this closed for now, we can open it when desirable.

**Comment by mherman22** [comment]
> > @kdaud how with this executable on Windows machine. Is there any work around concerning this issue
> 
> The changes need to be tested on a windows environment!
> 
> But will have this closed for now, we can open it when desirable.

i have actually tested those changes the whole of today, been trying to trouble shoot it but it seems to be throwing the same old error -> `Unknown lifecycle phase "and". You must specify a valid lifecycle phase or a goal in the format <plugin-prefix>:<goal> or <plugin-group-id>:<plugin-artifact-id>[:<plugin-version>]:<goal>. Available lifecycle phases are: validate, initialize, generate-sources, process-sources, generate-resources, process-resources, compile, process-classes, generate-test-sources, process-test-sources, generate-test-resources, process-test-resources, test-compile, process-test-classes, test, prepare-package, package, pre-integration-test, integration-test, post-integration-test, verify, install, deploy, pre-clean, clean, post-clean, pre-site, site, post-site, site-deploy. -> [Help 1]`

**Comment by kdaud** [comment]
Are you running windows 10 or 11 os?

**Comment by mherman22** [comment]
> @mherman22 are you running windows 10 or 11 os?

windows 10

**Comment by kdaud** [comment]
Are you able to share the logs via pastebin.com?

**Comment by mherman22** [comment]
> Are you able to share the logs via [pastebin](https://pastebin.com)?

when i trigger `npm run refapp2ClinicalVisit` i get this- https://pastebin.com/SX0RE7hY...

**Comment by kdaud** [comment]
From what you have shared, the os is able to install the project dependencies. 

I suggest you confirm whether node and npm are well installed on the machine. This might be root cause of the issue and logs when running the tests might not be in position to point this out!

**Comment by mherman22** [comment]
> Does the result remain the same when you check the version in a new terminal not within the project folder?

yes it does...


```
C:\Users\HERMAN MUHEREZA>node -v
v16.14.0

C:\Users\HERMAN MUHEREZA>npm -v
8.3.1

C:\Users\HERMAN MUHEREZA>
```

**Comment by kdaud** [comment]
@mherman22 do you mind to looking into this issue deeply and investigate why that error is thrown when running the tests?

**Comment by mherman22** [comment]
> @mherman22 what happens when you append `-e` the command to get a full stack trace log
> 
> ```js
> npm run refapp2ClinicalVisit -e
> ```

--> nearly the same result https://pastebin.com/CWjEGiCy

**Comment by mherman22** [comment]
> > @mherman22 what happens when you append `-e` the command to get a full stack trace log
> > ```js
> > npm run refapp2ClinicalVisit -e
> > ```
> 
> --> nearly the same result https://pastebin.com/CWjEGiCy

please take a look at [this which i get when i run mvn clean install ](https://pastebin.com/) against `\qaframework-bdd-tests` folder. to highlight it out there's this -> `Caused by: java.lang.AssertionError: The driver executable must exist: C:\Users\HERMAN%20MUHEREZA\Documents\GitHub\openmrs-contrib-qaframework\qaframework-bdd-tests\target\test-classes\firefoxdriver\linux\geckodriver`.

I have the same when i get chromedriver...!!!

**Comment by kdaud** [comment]
Could you share the content of `test.roperties` files?

**Comment by sherrif10** [comment]
Thanks @kdaud @mherman22  for sticking with this. I haven't seen the majority using Windows. My recommendation is to use Linux for better simulation. I think we are to agree with it since we don't have a goood experience on simulating with windows

**Comment by mherman22** [comment]
> Could you share the content of `test.roperties` files?

```
webapp.url=https://qa-refapp.openmrs.org/openmrs
login.username=admin
login.password=Admin123
login.location=Pharmacy
webdriver=firefox
login.auto=true
headless=false
webdriver.gecko.driver=firefoxdriver/linux/geckodriver
includes.csrftoken=true
```

**Comment by mherman22** [comment]
> I haven't seen the majority using Windows. My recommendation is to use Linux for better simulation. I think we are to agree with it since we don't have a goood experience on simulating with windows

you're right @sherrif10 , we just didnt want to go down without a fight 😄

**Comment by mherman22** [comment]
> Which windows os version are you testing this on?

Windows 10

**Comment by kdaud** [comment]
Will open it when needed.

**Pull request closed without merging**